cql3/statements/modification_statement.cc

```diff
@@ -316,6 +316,44 @@ modification_statement::execute_without_condition(query_processor& qp, service::
     });
 }
 
+namespace {
+future<::shared_ptr<cql_transport::messages::result_message>>
+process_forced_rebounce(const unsigned shard, query_processor& qp, const query_options& options) {
+    static std::atomic<int64_t> counter = {0};
+    static logging::logger logger("modification_statement");
+    if (counter.load(std::memory_order_acquire) <= 0) {
+        const auto counter_opt = utils::get_local_injector().inject_parameter<decltype(counter)::value_type>("forced_bounce_to_shard_counter");
+        decltype(counter)::value_type counter_value = 0;
+        if (!counter_opt) {
+            logger.warn("forced_bounce_to_shard_counter is not set. Using default value 1.");
+        } else {
+            try {
+                counter_value = boost::lexical_cast<decltype(counter_value)>(*counter_opt);
+            } catch (const boost::bad_lexical_cast& e) {
+                logger.warn("Incorrect forced_bounce_to_shard_counter value: [{}]. Using default value 1.", *counter_opt);
+            }
+        }
+        if (counter_value <= 0) {
+            counter_value = 1;
+        }
+        counter.store(counter_value, std::memory_order_release);
+    }
+
+    const auto prev_counter_value = counter.load(std::memory_order_acquire);
+    if (prev_counter_value <= 1) {
+        logger.info("Disabling forced_bounce_to_shard_counter.");
+        co_await utils::error_injection_type::disable_on_all("forced_bounce_to_shard_counter");
+        counter.store(0);
+    } else {
+        counter.fetch_sub(1, std::memory_order_acq_rel);
+    }
+
+    logger.info("Applying forced_bounce_to_shard_counter.");
+    co_return co_await make_ready_future<shared_ptr<cql_transport::messages::result_message>>(
+        qp.bounce_to_shard(shard, std::move(const_cast<cql3::query_options&>(options).take_cached_pk_function_calls())));
```

Comment:
So we always bounce to the same shard? Don't we want to move between shards on each bounce?

Reply:
Yes, I use the same shard, since this still reproduces multiple re-bouncing issue.